acceptance/api_v1_applications_test.go

```diff
@@ -34,139 +27,250 @@ var _ = Describe("API Application Endpoints", func() {
+				response, err := Curl("GET", fmt.Sprintf("%s/api/v1/orgs/%s/applications/bogus", serverURL, org), strings.NewReader(""))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(response).ToNot(BeNil())
+
+				defer response.Body.Close()
+				bodyBytes, err := ioutil.ReadAll(response.Body)
+				Expect(response.StatusCode).To(Equal(http.StatusNotFound), string(bodyBytes))
+			})
 		})
 
-		It("returns a 404 when the app does not exist", func() {
-			response, err := Curl("GET", fmt.Sprintf("%s/api/v1/orgs/%s/applications/bogus", serverURL, org), strings.NewReader(""))
-			Expect(err).ToNot(HaveOccurred())
-			Expect(response).ToNot(BeNil())
+		Describe("DELETE api/v1/orgs/:org/applications/:app", func() {
+			var service string
+
+			BeforeEach(func() {
+				service = newServiceName()
+				makeCustomService(service)
+				bindAppService(app1, service, org)
+			})
 
-			defer response.Body.Close()
-			bodyBytes, err := ioutil.ReadAll(response.Body)
-			Expect(response.StatusCode).To(Equal(http.StatusNotFound), string(bodyBytes))
+			AfterEach(func() {
+				deleteApp(app2) // This one was not deleted in the test
+				cleanupService(service)
+			})
+
+			It("removes the application, unbinds bound services", func() {
+				response, err := Curl("DELETE", fmt.Sprintf("%s/api/v1/orgs/%s/applications/%s", serverURL, org, app1), strings.NewReader(""))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(response).ToNot(BeNil())
+				defer response.Body.Close()
+				Expect(response.StatusCode).To(Equal(http.StatusOK))
+				bodyBytes, err := ioutil.ReadAll(response.Body)
+				Expect(err).ToNot(HaveOccurred())
+
+				var resp map[string][]string
+				err = json.Unmarshal(bodyBytes, &resp)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(resp).To(HaveLen(1))
+				Expect(resp).To(HaveKey("UnboundServices"))
+				Expect(resp["UnboundServices"]).To(ContainElement(service))
+			})
+
+			It("returns a 404 when the org does not exist", func() {
+				response, err := Curl("DELETE", fmt.Sprintf("%s/api/v1/orgs/idontexist/applications/%s", serverURL, app1), strings.NewReader(""))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(response).ToNot(BeNil())
+
+				defer response.Body.Close()
+				bodyBytes, err := ioutil.ReadAll(response.Body)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(response.StatusCode).To(Equal(http.StatusNotFound), string(bodyBytes))
+			})
+
+			It("returns a 404 when the app does not exist", func() {
+				response, err := Curl("DELETE", fmt.Sprintf("%s/api/v1/orgs/%s/applications/bogus", serverURL, org), strings.NewReader(""))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(response).ToNot(BeNil())
+
+				defer response.Body.Close()
+				bodyBytes, err := ioutil.ReadAll(response.Body)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(response.StatusCode).To(Equal(http.StatusNotFound), string(bodyBytes))
+			})
 		})
 	})
 
-	Describe("DELETE api/v1/orgs/:org/applications/:app", func() {
-		var service string
+	Context("Orgs", func() {
```

Comment:
I think this belongs to another test file. This one is called `api_v1_applications_test.go` . We should better use a `api_v1_orgs_test.go` for all orgs related tests.

Reply:
That bogosity is on me. Apologies.

Reply:
Done.